### PR TITLE
Extradocs

### PR DESCRIPTION
**Pull request by omad**
### Reason for this pull request

Our documentation needs improvement, this is a small start.

### Proposed changes

- Add _plantuml_ diagrams showing some of the internal architecture when loading data
- Add some documentation on the the _bit-masking_ functionality for PQ like products
- Assorted small grammar and style fixes
- Also make a refactor in `api\core.py` in how we check whether threaded reads are available

 - [x] Tests passed
 - [x] Fully documented, including `docs/about/whats_new.rst` for all changes

### CODE REVIEW

File: docs/user/intro.rst [hunk 1]
```diff
@@ -23,16 +23,18 @@ Check out the :ref:`installation` for instructions on configuring and setting up
 Australian Users
 ================
 
-If you have a login with the :term:`NCI`, check out the :ref:`nci_usage_guide`
+If you have are lucky enough to have a login with the :term:`NCI`, check out the :ref:`nci_usage_guide`.
+Geoscience Australia have a Data Cube installation with the Australian Landsat archive,
+along with many other datasets, set up and ready to use.
```
Comment (andrewdhicks) [inline docs/user/intro.rst:28]: Should we remove the `nci_usage` page in favor of a link to http://geoscienceaustralia.github.io/digitalearthau/

File: docs/dev/developer.rst [hunk 2]
```diff
@@ -22,10 +26,34 @@ Writing Large Scale Workflows
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+How to Create Masks within code
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. currentmodule:: datacube.storage
+
+.. automethod:: masking.describe_variable_flags
+.. automethod:: masking.make_mask
```
Comment (andrewdhicks) [inline docs/dev/developer.rst:56]: Move to a separate page?

**Review by andrewdhicks — APPROVED** [review]
Looks good to me.
Added a couple of suggestions.

File: docs/dev/developer.rst (hunk 2 above)
Comment (omad) [reply docs/dev/developer.rst:56]: Probably, but I say defer this to be part of a larger docs restructure.

File: docs/user/intro.rst (hunk 1 above)
Comment (omad) [reply docs/user/intro.rst:28]: Definitely! In a new PR.

**Pull request merged**